### DISCUSSION TITLE
Audio not playing correctly in some areas

### DISCUSSION THREAD

**Discussion by simox-23** [Wohlstand/TheXTech#438] [Q&A] [answered]
_Played on TheXTech version 1.3.6 (win64) using Windows 10. (https://github.com/Wohlstand/TheXTech/releases/download/v1.3.6/super-mario-bros-x-thextech-v1.3.6-win64.7z)_

### Hi, I noticed that when entering the Misplaced Pipehammer Maze (level4.lvlx) (The Invasion 2) underground area, the SPC plays a little strange. However, in other areas, the same SPC plays correctly.

https://user-images.githubusercontent.com/80751487/197364259-013225cf-cc38-42ae-b06d-96f2ab2d2859.mp4
(SPC played wrong)

https://user-images.githubusercontent.com/80751487/197364293-72c83b7f-0563-40b0-923c-7783ad37b12c.mp4
(SPC played correctly)

I don't exactly know if this is caused by the reverb mode in different areas. But I can definitely tell that the SPC in the first video sounds off.

Thanks for reading and keep up the great work!

**Comment by Wohlstand** [comment, marked as answer]
The sounding of the song got been altered because I disabled native echo of that SPC file at the Sound FX effect's settings. That was done in a goal to prevent the conflict between custom echo and native echo of SPC file (this conflict will lead the double-echo and messy sounding). You can change the echo effect from SMW cave into SMB3 underground by editing of level4/sounds.ini and changing the name of the effect to use.

Note: the native SPC echo gets disabled into the episode-wide sounds.ini at evey declared effect.

**Comment by simox-23** [comment]
Ok, I saw now that warp5.lvlx uses the "bonusroom" type reverb, while level4.lvlx uses the "cave" type reverb. So it's normal that they're sounding different.
Thanks for the response! :)

**Comment by Wohlstand** [comment]
> So it's normal that they're sounding different.

Yes, that's normal. And so, the specific sounding of the bass in the "underground" song is a result of native echo enabled. If you disable the echo via hacks while playing the game, you'll hear the same as level4 at me (except for the cave echo).

Note: the effect is not reverb, it's an echo. There are different effects:
- The echo is collecting a small sound buffer and repeats it with delaying. The echo algorithm used at me is fully repeats functionality of S-DSP chip used in SNES.
- The reverb is an echo (technically) too, but, with a very very short delay. Unlike "echo", the result of reverb is a "print" that sounds monotonous. For the reverb effect, the FreeVerb library is used at me.

**Comment by Wohlstand** [comment]
btw, I'll move this into "Discussions"
